cwl/src/main/scala/cwl/WorkflowStep.scala

```diff
@@ -60,14 +60,20 @@ case class WorkflowStep(
                      workflowInputs: Map[String, GraphNodeOutputPort]): Checked[Set[GraphNode]] = {
 
     // To avoid duplicating nodes, return immediately if we've already covered this node
-    val haveWeSeenThisStep: Boolean = knownNodes.collect { case TaskCallNode(identifier, _, _, _) => identifier }.contains(unqualifiedStepId)
+    val haveWeSeenThisStep: Boolean = knownNodes.collect {
+      case TaskCallNode(identifier, _, _, _) => identifier
+      case WorkflowCallNode(identifier, _, _, _) => identifier
+      // TODO CWL: Catch known ExpressionTools too
```

Comment:
good that there's at least a TODO now, but I like what you did before with exploding even better